router/internal/persistedoperation/apq/redis.go

```diff
@@ -0,0 +1,63 @@
+package apq
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/redis/go-redis/v9"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"go.uber.org/zap"
+	"time"
+)
+
+type RedisOptions struct {
+	Logger        *zap.Logger
+	StorageConfig *config.BaseStorageProvider
+	ApqConfig     *config.AutomaticPersistedQueriesConfig
+	Prefix        string
+}
+
+type redisClient struct {
+	logger *zap.Logger
+	client *redis.Client
+	prefix string
+}
+
+func NewRedisClient(opts *RedisOptions) (KVClient, error) {
+	if opts.StorageConfig == nil {
+		return nil, errors.New("storage config is nil")
+	}
+
+	options, err := redis.ParseURL(opts.StorageConfig.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the redis connection url: %w", err)
+	}
+
+	innerClient := redis.NewClient(options)
+
+	rclient := &redisClient{
+		logger: opts.Logger,
+		client: innerClient,
+		prefix: opts.Prefix,
+	}
+
+	return rclient, nil
+}
+
+func (r *redisClient) Get(ctx context.Context, clientName, operationHash string) ([]byte, error) {
+	cmd := r.client.Get(ctx, r.prefix+operationHash)
+	if errors.Is(cmd.Err(), redis.Nil) {
```

Comment:
In what scenario does this happen?

Reply:
If it's a cache miss from redis, it returns an error of `redis.Nil`. In that scenario, we don't want to return an error because it could be that there is a top-level query that was attached to the request

Reply:
thanks